File: IntersectionOfTwoLinkedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */

public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
    ListNode cur = headA;
    ListNode curO = headB;
    
    while( cur != curO){
        cur = cur == null? headB : cur.next;
        curO = curO == null? headA : curO.next;    
    }
    
    return cur;

    }
}

File: IntersectionOfTwoLinkedList.js
class ListNode {
    constructor(x) {
        this.val = x;
        this.next = null;
    }
}

class Solution {
    getIntersectionNode(headA, headB) {
        let cur = headA;
        let curO = headB;

        while (cur !== curO) {
            cur = cur === null ? headB : cur.next;
            curO = curO === null ? headA : curO.next;
        }

        return cur;
    }
}

